core/templates/pages/exploration-player-page/modals/display-hint-modal.component.ts

```diff
@@ -33,12 +34,12 @@ import { PlayerTranscriptService } from '../services/player-transcript.service';
   templateUrl: './display-hint-modal.component.html'
 })
 export class DisplayHintModalComponent {
-  index: number;
-  COMPONENT_NAME_HINT: string;
-  hint: SubtitledHtml;
-  displayedCard: StateCard;
-  recordedVoiceovers: RecordedVoiceovers;
-  hintContentId: string;
+  index: number | undefined;
+  COMPONENT_NAME_HINT: string = AppConstants.COMPONENT_NAME_HINT;
+  hint!: SubtitledHtml | null;
```

Comment:
Why is `null` allowed here? Also why we need `!`?

Reply:
We need ! here because it's undefined before calling `ngOnInit()` function. `null` is allowed here for hint because displayHint() funtion in [hints-and-solution-manager.service.ts file](https://github.com/oppia/oppia/blob/03bb986c0374f7412fbff94119b1081014b8884e/core/templates/pages/exploration-player-page/services/hints-and-solution-manager.service.ts) can return `null`.